sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateSampleDao.java

```diff
@@ -346,21 +347,35 @@ public boolean aliasExists(String alias) throws IOException {
 
   @Override
   public Collection<Sample> listBySearch(String querystr) throws IOException {
-    Criteria criteria = currentSession().createCriteria(Sample.class);
-    criteria.add(Restrictions.or(Restrictions.ilike("identificationBarcode", "%" + querystr + "%"),
-        Restrictions.ilike("name", "%" + querystr + "%"), Restrictions.ilike("alias", "%" + querystr + "%")));
+    Criteria criteria = currentSession().createCriteria(SampleImpl.class);
+    criteria.add(Restrictions.or(
+        Restrictions.ilike("identificationBarcode", "%" + querystr + "%"),
+        Restrictions.ilike("name", "%" + querystr + "%"),
+        Restrictions.ilike("alias", "%" + querystr + "%")));
     @SuppressWarnings("unchecked")
     List<Sample> records = criteria.list();
     return fetchSqlStore(records);
   }
 
   @Override
+  public Long countBySearch(String querystr) throws IOException {
+    Criteria criteria = currentSession().createCriteria(SampleImpl.class);
+    criteria.add(Restrictions.or(
+        Restrictions.ilike("identificationBarcode", "%" + querystr + "%"),
+        Restrictions.ilike("name", "%" + querystr + "%"),
+        Restrictions.ilike("alias", "%" + querystr + "%")));
+    return (Long) criteria.setProjection(Projections.rowCount()).uniqueResult();
+  }
+
```

Comment:
Would :heart:  to see a test for this new method and perhaps an `@CoverageIgnore` for the trivial implementation in the service layer :neckbeard: